main/src/main/java/cgeo/geocaching/connector/gc/ImportBookmarkLinks.java

```diff
@@ -9,13 +9,90 @@
 import android.net.Uri;
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 
 import java.util.regex.Pattern;
 
+import javax.annotation.RegEx;
+
 public class ImportBookmarkLinks extends AppCompatActivity {
 
+    private final UrlToIdParser intentUrlParser = defaultBookmarkListUrlToIdParser();
+
+    public static final String BOOKMARK_LIST_API_PREFIX =
+            "https://www.geocaching.com/plan/api/gpx/list/";
+
+    interface UrlToIdParser {
+        @Nullable String tryExtractFromIntentUrl(@Nullable String intentUrl);
+    }
+
+    static class SingleUrlToIdParser implements UrlToIdParser {
+        private final Pattern matcherPattern;
+        // Android API 21 apparently does not support named capturing groups in Pattern
+        // which has introduced in java 1.7...
+        private final int groupToExtract;
+
+        SingleUrlToIdParser(
+                @RegEx @NonNull final String regex,
+                final int groupToExtract
+        ) {
+            this.matcherPattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+            this.groupToExtract = groupToExtract;
+        }
+
+        @Override
+        public String tryExtractFromIntentUrl(final String intentUrl) {
+            if (intentUrl == null) {
```

Comment:
Instead of the logic in the method, use a call to `TextUtils.getMatch(@Nullable final String data, final Pattern pattern, final boolean trim, final int group, @Nullable final String defaultValue, final boolean last)`